__tests__/unit/fact.test.js

```diff
@@ -0,0 +1,415 @@
+  // trivia and math share the same prefix
+  test("When wantFragment is true, returns partial text only", function () {
+    const sentence = getSentence({
+      wantFragment: { fragment: true },
+      number: 88,
+      type: "trivia",
+      data: {
+        text: "the number of keys on a piano (36 black and 52 white)",
+      },
+    });
+    expect(sentence).toBe(
+      "the number of keys on a piano (36 black and 52 white)"
+    );
+  });
+
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 88,
+      type: "trivia",
+      data: {
+        text: "the number of keys on a piano (36 black and 52 white)",
+      },
+    });
+    expect(sentence).toBe(
+      "88 is the number of keys on a piano (36 black and 52 white)."
+    );
+  });
+
+  test("When trivia number is invalid, returns sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 88888,
+      type: "trivia",
+      data: {
+        text: "an unremarkable number",
+      },
+    });
+    expect(sentence).toBe("88888 is an unremarkable number.");
+  });
+});
+
+describe("getSentence() for type 'date' ", function () {
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 1 / 1,
+      type: "date",
+      data: {
+        text: "that Victoria is crowned princess of Sweden",
+        year: 1980,
+      },
+    });
+    expect(sentence).toBe(
+      "January 1st is the day in 1980 that that Victoria is crowned princess of Sweden."
+    );
+  });
+});
+
+describe("getDefaultMsg() for all 4 types", function () {
+  test("random default msg for type:'year'", function () {
+    const yearMsgs = [
+      "9999999999 is the year that nothing remarkable happened.",
+      "9999999999 is the year that the Earth probably went around the Sun.",
+      "9999999999 is the year that nothing interesting came to pass.",
+      "9999999999 is the year that we do not know what happened.",
+    ];
+    const sentence = getDefaultMsg({
+      number: 9999999999,
+      type: "year",
+      options: {},
+    });
+    expect(yearMsgs.includes(sentence)).toEqual(true);
+  });
+
+  test("random default msg for type:'math' & type:'trivia'", function () {
+    const mathMsgs = [
+      "9999999999 is an uninteresting number.",
+      "9999999999 is a boring number.",
+      "9999999999 is an unremarkable number.",
+      "9999999999 is a number for which we're missing a fact (submit one to numbersapi at google mail!).",
+    ];
+    const sentence = getDefaultMsg({
+      number: 9999999999,
+      type: "math",
+      options: {},
+    });
+    expect(mathMsgs.includes(sentence)).toEqual(true);
+  });
+
+  test("default msg for type:'date' with no fact", function () {
+    const sentence = getDefaultMsg({
+      number: 1 / 1,
```

Comment:
same comment as above, maybe i don't quite understand how the API works with dates